aeon/dj_pipeline/analysis/block_analysis.py

```diff
@@ -309,18 +309,23 @@ def make(self, key):
         # Get all unique subjects that visited the environment over the entire exp;
         # For each subject, see 'type' of visit most recent to start of block
         # If "Exit", this animal was not in the block.
+        subject_names = (acquisition.Experiment.Subject & key).fetch("subject", order_by="subject").tolist()
+        if not len(subject_names):
+            raise ValueError(
+                "No subjects found in the `acquisition.Experiment.Subject`, missing a manual insert step?."
+            )
+
         subject_visits_df = fetch_stream(
             acquisition.Environment.SubjectVisits
             & key
             & f'chunk_start <= "{chunk_keys[-1]["chunk_start"]}"'
         )[:block_start]
         subject_visits_df = subject_visits_df[subject_visits_df.region == "Environment"]
-        subject_visits_df = subject_visits_df[~subject_visits_df.id.str.contains("Test", case=False)]
-        subject_names = []
-        for subject_name in set(subject_visits_df.id):
+
+        for subject_name in subject_names:
```

Comment:
```suggestion
        last_environment_visit = (
            subject_visits_df[subject_visits_df.region == "Environment"].groupby("id").tail(1)
        )
        environment_exit = last_environment_visit[last_environment_visit.type == "Exit"]
        subject_names = set(subject_visits_df.id) - set(environment_exit.id)
```
I think we can do away with the for loop and remove [L326-238](https://github.com/SainsburyWellcomeCentre/aeon_mecha/blob/50da39d2769337cee9c82a56e1186c3c5b9980b3/aeon/dj_pipeline/analysis/block_analysis.py#L326-L328)

Reply:
But you mean `subject_names = set(subject_names) - set(environment_exit.id)`, right?

Reply:
ah yes sorry.